modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +86,41 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = enum_zones
+      weight = 0 # sub zones might be independent from similar named parent zones; use weight for longest suffix match
+      matched_zone = nil
+
+      dns_zones.each do |zone|
+        zone_labels = zone.split(".").reverse
+        zone_weight = zone_labels.length
+        fqdn_labels = fqdn.split(".")
+        fqdn_labels.shift
+        is_match = zone_labels.all? { |zone_label| zone_label == fqdn_labels.pop }
+        # match only the longest zone suffix
+        if is_match && zone_weight >= weight
+          matched_zone = zone
+          weight = zone_weight
+        end
+      end
+      unless matched_zone
+        logger.error "The DNS server has no authoritative zone for #{fqdn}!"
```

Comment:
This should raise a Proxy::Dns::NotFound exception instead. No need for logging it either, the main plugin will do that instead.